data_model/src/query.rs

```diff
@@ -1005,24 +1005,24 @@ pub mod trigger {
     }
 
     impl Query for FindAllActiveTriggerIds {
-        type Output = Vec<<Trigger<FilterBox> as Identifiable>::Id>;
+        type Output = Vec<<Trigger<FilterBox, Executable> as Identifiable>::Id>;
     }
 
     query! {
         /// Find Trigger given its ID.
         #[derive(Display)]
         #[display(fmt = "Find `{id}` trigger")]
         #[repr(transparent)]
-        // SAFETY: `FindTriggerById` has no trap representation in `EvaluatesTo<<Trigger<FilterBox> as Identifiable>::Id>`
+        // SAFETY: `FindTriggerById` has no trap representation in `EvaluatesTo<<Trigger<FilterBox, Executable> as Identifiable>::Id>`
         #[ffi_type(unsafe {robust})]
         pub struct FindTriggerById {
             /// The Identification of the trigger to be found.
-            pub id: EvaluatesTo<<Trigger<FilterBox> as Identifiable>::Id>,
+            pub id: EvaluatesTo<<Trigger<FilterBox, Executable> as Identifiable>::Id>,
         }
     }
 
     impl Query for FindTriggerById {
-        type Output = Trigger<FilterBox>;
+        type Output = Trigger<FilterBox, Option<Vec<Instruction>>>;
```

Comment:
> It's not as if this information is lost, it's on the blockchain.

I also thought about that and realized, that we respond only with information about current world state. We won't respond with deleted account even if it existed before.

Same for WASM inside triggers. It exists in block history, so technically it's possible to retrieve this info somehow. But our queries works only with current state. And raw WASM is never presented in the state.

I think, we should handle this problem in another way. With some kind of history queries

Reply:
Also in 99% there is no meaning to query WASM, because you probably can't do anythin useful with it

Reply:
And storing useless raw WASM will just take a lot of space. And that only for such strange queires

Reply:
> Also in 99% there is no meaning to query WASM, because you probably can't do anything useful with it

There is, for Iroha explorer. Although it's not an easy task to display it in a readable way, it is possible to transform it to text and show to an explorer user.

Reply:
> There is, for Iroha explorer. Although it's not an easy task to display it in a readable way, it is possible to transform it to text and show to an explorer user.

is it really useful? because you cannot get rust source code, you can only get wasm binary or `wat` format at best which is hardly usable IMO

Reply:
> We won't respond with deleted account even if it existed before.

IMO situation with account is a bit different because account deletion will be on chain.

And in case of trigger difference is that we store trigger differently between block store and world. 

Current return type of query is kinda confusing to me, but searching through blocks or store original wasm doesn't seems to be efficient.

Reply:
So should we gather opinions from the SDK devs?

Reply:
After a discussion on todays status we decided to return optimized WASM